test/JoinTest.cpp

```diff
@@ -0,0 +1,429 @@
+// Copyright 2015, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Björn Buchhold (buchhold@informatik.uni-freiburg.de)
+// Co-Author: Andre Schlegel (November of 2022, schlegea@informatik.uni-freiburg.de)
+
+#include <gtest/gtest.h>
+
+#include <cstdio>
+#include <fstream>
+#include <sstream>
+#include <algorithm>
+#include <tuple>
+
+#include "engine/CallFixedSize.h"
+#include "engine/Engine.h"
+#include "engine/Join.h"
+#include "engine/OptionalJoin.h"
+#include "util/Random.h"
+#include "engine/QueryExecutionTree.h"
+#include "util/Forward.h"
+#include "util/SourceLocation.h"
+#include "./util/GTestHelpers.h"
+#include "engine/idTable/IdTable.h"
+
+ad_utility::AllocatorWithLimit<Id>& allocator() {
+  static ad_utility::AllocatorWithLimit<Id> a{
+      ad_utility::makeAllocationMemoryLeftThreadsafeObject(
+          std::numeric_limits<size_t>::max())};
+  return a;
+}
+
+auto I = [](const auto& id) {
+  return Id::makeFromVocabIndex(VocabIndex::make(id));
+};
+
+// For easier reading. We repeat that type combination so often, that this
+// will make things a lot easier in terms of reading and writing.
+using vectorTable = std::vector<std::vector<size_t>>;
+
+/*
+ * Return an 'IdTable' with the given 'tableContent'. all rows must have the
+ * same length.
+*/
+IdTable makeIdTableFromVector(vectorTable tableContent) {
+  AD_CHECK(!tableContent.empty());
+  IdTable result{tableContent[0].size(), allocator()};
+
+  // Copying the content into the table.
+  for (const auto& row: tableContent) {
+    AD_CHECK(row.size() == result.numColumns()); // All rows of an IdTable must have the same length.
+    const size_t backIndex{result.size()};
+    result.emplace_back();
+
+    for (size_t c = 0; c < row.size(); c++) {
+      result(backIndex, c) = I(row[c]);
+    }
+  }
+
+  return result;
+}
+
+/*
+ * Does what it says on the tin: Save an IdTable with the corresponding
+ * join column.
+ */
+struct IdTableAndJoinColumn {
+  IdTable idTable;
+  size_t joinColumn;
+};
+
+/*
+ * A structure containing all information needed for a normal join test. A
+ * normal join test is defined as two IdTables being joined, the resulting
+ * IdTable tested, if it is sorted after the join column, or not, and this
+ * IdTable is than compared with the expectedResult.
+ */
+struct normalJoinTest {
+  IdTableAndJoinColumn leftInput;
+  IdTableAndJoinColumn rightInput;
+  IdTable expectedResult;
+  bool resultMustBeSortedByJoinColumn;
```

Comment:
If we want to check, if the result is sorted after the join column, yes.

Detailed explanation: The function that tests all join algorithms currently does that by manipulating the test cases and calling over helper functions, that then 'execute' the test cases.
These helper functions only check, if the result is sorted, if `resultMustBeSortedByJoinColumn` is true.
I could change it, so that the testing for being sorted is done in `runTestCasesForAllJoinAlgorithm`, but I don't really see a reason for doing so.